src/main/scala/jio/package.scala

```diff
@@ -29,19 +27,45 @@ package object jio extends JioFiles {
   }
 
   implicit class PathOps(val p: Path) extends AnyVal {
+
     def /(name: String): Path = p resolve name
 
     def mkdir(attrs: AnyFileAttr*): Path = createDirectory(p, attrs: _*)
     def exists                           = p.toFile.exists
-    def isSymbolicLink()                 = Files isSymbolicLink p
-    def readSymbolicLink()               = Files readSymbolicLink p
-    def readAllBytes()                   = Files readAllBytes p
-    def list(): Vector[Path]             = Files list p toVector
+    def isFile                           = Files.isRegularFile(p, NOFOLLOW_LINKS)
+    def isDirectory                      = Files.isDirectory(p, NOFOLLOW_LINKS)
+    def isSymbolicLink                   = Files isSymbolicLink p
+    def readSymbolicLink                 = Files readSymbolicLink p
+    def allBytes                         = Files readAllBytes p
+    def list: Vector[Path]               = (Files list p).toVector
     def filename: String                 = p.getFileName.toString
+    def blockSize: Long                  = 512 // FIXME
+    def blockCount: Long                 = (attributes.size + blockSize - 1) / blockSize
+    def size: Long                       = attributes.size
+    def atime: Long                      = attributes.lastAccessTime.toMillis
+    def mtime: Long                      = attributes.lastModifiedTime.toMillis
 
     def openChannel(opts: OpenOption*): FileChannel = jnc.FileChannel.open(p, opts: _*)
+
+    def permissions: PosixFilePermissions = {
+      val pfp = (Files getPosixFilePermissions (p, NOFOLLOW_LINKS)).asScala
+      import jnfa.PosixFilePermission._
+      PosixFilePermissions(
+        pfp(GROUP_READ) , pfp(GROUP_WRITE) , pfp(GROUP_EXECUTE),
+        pfp(OWNER_READ) , pfp(OWNER_WRITE) , pfp(OWNER_EXECUTE),
+        pfp(OTHERS_READ), pfp(OTHERS_WRITE), pfp(OTHERS_EXECUTE)
+      )
+    }
+
+    def attributes: BasicFileAttributes = Files readAttributes(p, classOf[BasicFileAttributes], NOFOLLOW_LINKS)
   }
 
```

Comment:
Here's some permission related code from my unpublished work, for whatever it may be worth. I remember the matter being very resistant to an appealing interface.
```scala
object Uid {
  def apply(s: StructStat): Uid = Uid(s.st_uid.toInt)
  def apply(attrs: PosixFileAttributes): Uid = attrs match {
    case NoFileAttrs => JvmUid
    case _           => Reflection uid attrs
  }
}
object Gid {
  def apply(s: StructStat): Gid = Gid(s.st_gid.toInt)
  def apply(attrs: PosixFileAttributes): Gid = attrs match {
    case NoFileAttrs => JvmGid
    case _           => Reflection gid attrs
  }
}

final case class Uid(value: Int) extends AnyVal
final case class Gid(value: Int) extends AnyVal
final case class Access(mask: Int) extends AnyVal {
  def | (that: Access): Access = Access(mask | that.mask)
  override def toString = Access to_s mask
}

object Access {
  private def to_s(bits: Int): String =
    ("rwxrwxrwx".toCharArray, 8 to 0 by -1).zipped map ((c, i) => if ((bits & (1 << i)) == 0) '-' else c) mkString ""

  private def create(bits: Long) = Access(bits.toInt & ((1 << 9) - 1))

  def apply(mode: TypeMode): Access             = create(mode.longValue)
  def apply(stat: StructStat): Access           = create(stat.st_mode)
  def apply(p: Path): Access                    = apply(p.attrs)
  def apply(attrs: PosixFileAttributes): Access = toBits(attrs)

  private def toBit(p: PosixFilePermission): Int = p match {
    case OWNER_READ     => 1 << 8
    case OWNER_WRITE    => 1 << 7
    case OWNER_EXECUTE  => 1 << 6
    case GROUP_READ     => 1 << 5
    case GROUP_WRITE    => 1 << 4
    case GROUP_EXECUTE  => 1 << 3
    case OTHERS_READ    => 1 << 2
    case OTHERS_WRITE   => 1 << 1
    case OTHERS_EXECUTE => 1 << 0
  }
  def toBits(p: PosixFileAttributes): Access  = Access(p.permissions.asScala.foldLeft(0)((res, x) => res | toBit(x)))
  def toBits(ps: PosixFilePermission*): Access = Access(ps.foldLeft(0)((res, x) => res | toBit(x)))

  private def read  = Seq(OWNER_READ, GROUP_READ, OTHERS_READ)
  private def write = Seq(OWNER_WRITE, GROUP_WRITE, OTHERS_WRITE)
  private def exec  = Seq(OWNER_EXECUTE, GROUP_EXECUTE, OTHERS_EXECUTE)

  def r   = toBits(read: _*)
  def w   = toBits(write: _*)
  def x   = toBits(exec: _*)
  def rx  = r | x
  def rw  = r | w
  def rwx = r | w | x

  def `444` = r
  def `555` = rx
  def `666` = rw
  def `777` = rwx
  def `664` = toBits(read ++ write filterNot (_ == OTHERS_WRITE): _*)
  def `775` = toBits(read ++ write ++ exec filterNot (_ == OTHERS_WRITE): _*)
}
```